traittypes/traittypes.py

```diff
@@ -206,3 +202,67 @@ def __init__(self, default_value=Empty, allow_none=False, dtype=None, **kwargs):
             kwargs['klass'] = pd.Series
         super(Series, self).__init__(
             default_value=default_value, allow_none=allow_none, dtype=dtype, **kwargs)
+
+
+class XarrayType(SciType):
+
+    """An xarray dataset trait type."""
+
+    info_text = 'an xarray dataset'
+
+    klass = None
+
+    def validate(self, obj, value):
+        if value is None and not self.allow_none:
+            self.error(obj, value)
+        if value is None or value is Undefined:
+            return super(XarrayType, self).validate(obj, value)
+        try:
+            value = self.klass(value)
+        except (ValueError, TypeError) as e:
+            raise TraitError(e)
+        return super(XarrayType, self).validate(obj, value)
+
+    def set(self, obj, value):
+        new_value = self._validate(obj, value)
+        old_value = obj._trait_values.get(self.name, self.default_value)
+        obj._trait_values[self.name] = new_value
+        if ((old_value is None and new_value is not None) or
+                (old_value is Undefined and new_value is not Undefined) or
+                not old_value.equals(new_value)):
+            obj._notify_trait(self.name, old_value, new_value)
+
+    def __init__(self, default_value=Empty, allow_none=False, dtype=None, klass=None, **kwargs):
+        if klass is None:
+            klass = self.klass
+        if (klass is not None) and inspect.isclass(klass):
+            self.klass = klass
+        else:
+            raise TraitError('The klass attribute must be a class'
+                                ' not: %r' % klass)
+        self.dtype = dtype
```

Comment:
I think that the `dtype` only makes sense in the `DataArray`  but not in the `Dataset` case.

- Maybe it is a good occasion to also remove it from the `PandasType`, and only keep it in the `Series`.
- Just like `Array` has `dtype = None` at the class level, `Series` and `DataArray` should probably too.